update_shotgun_users.py

```diff
@@ -0,0 +1,158 @@
+
+        # Email has already been mapped to a JIRA user, so skip it.
+        if user["email"] in mapped_emails:
+            logger.info(
+                "The email '{}' from '{}' has already been associated with a JIRA account.".format(
+                    user["email"], user["login"]
+                )
+            )
+            continue
+
+        # Let's try to find a JIRA user associated with that email for the given JIRA project.
+        jira_user = jira.find_jira_assignee_for_issue(
+            user["email"], jira_project=project
+        )
+        # If no user want found, let the user know.
+        if jira_user is None:
+            continue
+
+        # A JIRA user was found, so let's update Shotgun with it's accountId!
+        sg.update("HumanUser", user["id"], {"sg_jira_account_id": jira_user.accountId})
+        logger.info(
+            "Shotgun user '{}' ('{}') has been matched to a JIRA user with the same email.".format(
+                user["login"], user["email"]
+            )
+        )
+        # Keep track of this new mapped email so that if we encounter it again we don't try to map
+        # that user again.
+        mapped_emails.add(user["email"])
+
+
+def _get_settings():
+    """
+    Retrieves the parameters necessary to run the app, i.e. logging settings and credentials for
+    both JIRA and Shotgun.
+
+    :returns: Tuple of (log settings, )
+    """
+    # Parse the commend line
+    parser = argparse.ArgumentParser(
+        description="Matches Shotgun users with JIRA users for JIRA Cloud.",
+        epilog=(
+            "This script will match the first Shotgun user with a given email with the "
+            "associated JIRA user. If for some reason there are multiple Shotgun users "
+            "with the same email, you can go back in Shotgun and reassign the "
+            "sg_jira_account_id value to the right Shotgun user."
+        ),
+    )
+    parser.add_argument("--settings", help="Full path to settings file.", required=True)
+
+    parser.add_argument(
+        "--project",
+        type=str,
+        help="The key of a JIRA project that will be synced with Shotgun.",
+    )
+    args = parser.parse_args()
+
+    logger_settings, shotgun_settings, jira_settings, _ = Bridge.read_settings(
+        args.settings
+    )
+    return logger_settings, shotgun_settings, jira_settings, args.project
+
+
+def main():
+    """
+    Map Shotgun users to JIRA users using their respective emails.
+    """
+    logger_settings, shotgun_settings, jira_settings, project = _get_settings()
+
+    # Apply settings
+    if logger_settings:
+        logging.config.dictConfig(logger_settings)
+
+    # Set the logger settings first since JIRA session is chatty.
+    jira = JiraSession(
+        jira_settings["site"],
+        basic_auth=(jira_settings["user"], jira_settings["secret"]),
+    )
+
+    if not jira.is_jira_cloud:
+        logger.error("This script can be run for JIRA Cloud only.")
+        return
```

Comment:
How about a non-zero int return here (and in other error situations, plus a `return 0` on success) and a `sys.exit(main())` at the bottom?

Reply:
I'm just thinking of someone integrating a call to this script in some other tool that creates new user accounts or something. Knowing whether the process succeeded or not via exit status is useful.

Reply:
Yes! Excellent point. I'll add it right away.